l10n_be_vat_reports/wizard/l10n_be_partner_vat_listing.py

```diff
@@ -0,0 +1,371 @@
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import time
+import base64
+
+from openerp import fields, models, api, _
+from openerp.report import report_sxw
+from openerp.exceptions import Warning
+
+
+class VATListingClients(models.TransientModel):
+    _name = 'vat.listing.clients'
+
+    name = fields.Char('Client Name')
+    vat = fields.Char('VAT')
+    turnover = fields.Float('Base Amount')
+    vat_amount = fields.Float('VAT Amount')
+
+
+class PartnerVAT(models.TransientModel):
+    """ Vat Listing """
+    _name = "partner.vat"
+
+    @api.multi
+    def get_partner(self):
+        context = dict(self.env.context)
+        obj_partner = self.env['res.partner']
+        obj_vat_lclient = self.env['vat.listing.clients']
+        obj_model_data = self.env['ir.model.data']
+        data = self.read()[0]
+        year = data['year']
+        date_start = year + '-01-01'
+        date_stop = year + '-12-31'
+
+        partners = obj_vat_lclient.browse([])
+        be_partners = obj_partner.search([('vat', 'ilike', 'BE%')])
+        if not be_partners:
+            raise Warning(_('No belgium contact with a VAT number '
+                            'in your database.'))
+        query = """
+            SELECT sub1.partner_id, sub1.name, sub1.vat,
+                   COALESCE(sub1.turnover, 0) as turnover,
+                   COALESCE(sub2.vat_amount, 0) as vat_amount
+            FROM (SELECT l.partner_id, p.name, p.vat,
+                         SUM(CASE WHEN tag_xmlid.name = 'tax_tag_49'
+                             THEN -ABS(l.balance)
+                             ELSE ABS(l.balance) END)
+                         as turnover
+                  FROM account_move_line l
+                  LEFT JOIN res_partner p ON l.partner_id = p.id
+                  LEFT JOIN account_tax t ON l.tax_line_id = t.id
+                  LEFT JOIN account_tax_account_tag tagsrel
+                      ON t.id = tagsrel.account_tax_id
+                  LEFT JOIN ir_model_data tag_xmlid ON (
+                      tag_xmlid.model = 'account.account.tag'
+                      AND tagsrel.account_account_tag_id = tag_xmlid.res_id)
+                  WHERE tag_xmlid.name IN (
+                      'tax_tag_00', 'tax_tag_01', 'tax_tag_02',
+                      'tax_tag_03', 'tax_tag_45', 'tax_tag_49')
+                  AND l.partner_id IN %s
+                  AND l.date BETWEEN %s AND %s
+                  GROUP BY l.partner_id, p.name, p.vat) AS sub1
```

Comment:
@andreparames the turnover computation is incorrect. 

In v9+, we must sum move lines where "tax_ids has one of (00,01,02,03,45,49)" to find the tax bases.

Reply:
But that's what I do (in lines 78-83), it's just that I search using the XML-IDs. I find all the account_tax with a tag that has on of the following XML-IDs: 'tax_tag_00', 'tax_tag_01', 'tax_tag_02', 'tax_tag_03', 'tax_tag_45', 'tax_tag_49', which are defined in the v9 of l10n_be: https://github.com/odoo/odoo/blob/9.0/addons/l10n_be/account_tax_template.xml#L5